---
title: "Final Report: Your Title Here"
author: "Derek Newberger"
date: "November, 26 2018"
output:
    github_document: default
bibliography: references.bib
csl: bioinformatics.csl
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Add about 2-3 pages here. Across the whole manuscript, you should cite at least 20 peer reviewed articles.

Relevant abiotic factors in this study include characteristics of soil and chemistry of the organic compound within it. The study placed sample weights into three main categories, macroaggregates (250-2,000 μm and 63-250 μm), microaggregates (20-63μm and 2-20 μm), and clay particles (2 μm). In its broadest sense, a macroaggregates are simply relatively large particles. Similar studies have described macroaggregates as 250-2,000 μm clumps which form from fresh residues since it is a carbon source for microbes and encourages the production of micro-derived binding agents. Microaggregates are formed within the much larger macroaggregates and typically maintains stability even if the macroaggregate it is inside, breaks apart. Bothof these structures influence the C-N availability [@six2000soil]. The study defined clay particles as particles, lower than 2 μm, which is in line with most estimates and although not explicitly defined, the silty-clay particles can be defined as soil particles that are intermixed with paticles between 2-50  μm [@six2000soil:1]. 

# Methods

## Study design of Constancias et al. 2014

### Field soil sampling:

During springtime in Burgundy, France in the agricultural landscape of Fénay, soils were sampled from three plots of soil characterized by silty clay calcisol. The agricultural practices on the grassland plot, the tilled plot, and the stubble ploughed plot, of the past eight years were monitored. Each plot had minor differences in their parameters. For sampling, a geological reference point of one meter was used and soil cores were from the surface horizon (0-20 cm) and pooled. After being sieved (2mm), the samples were air dried, and then stored at 4 C until the fraction procedure.

### Soil fractionation procedure:

They used a gentle physical soil fraction method (Chotte et al. 2002), to isolate the soil microenvironments by size. A subsample of 50 g of 2-mm was shaken in 250 mL of Milli-Q water for 3 h. Using the two sieves (250- and 63-μm) particles and water-stable aggregates were separated by wet sieving in approximately 3 cm of water 60 times during 2 min.  The 2,000–250- and 250–63-μm fractions were macroaggregates associated to coarse and fine sands, respectively. Stoke’s law was used to separate the microaggregates with particles of silt and loam by sedimentation and siphoning. Water was removed as clay fractions of <2 μm were recovered by centrifuging at 11,000g, 4 °C for 20 min. Samples were then freeze-dried. This procedure was carried out in replicates. Replicates were lyophilized and stored at −40 °C, and the three subsamples were used for physicochemical and microbiological analysis.

### Soil physicochemical analysis

The lyophilized fractions were weighed, organic carbon and total nitrogen contents of samples were characterized according to NF ISO 10694 and 13878.  Soil texture of bulk soils was determined according to NF X 31–107. Size distribution of the particles and aggregates of each fraction was verified by laser granulometry (Mastersizer 2000, Malvern Ltd). Using adhesive graphite tabs, aluminum stubs gold-palladium (12–15 nm thick), Polaron SC 7620 Mini Sputter Coater, samples were observed with a Philips XL-30ESEM LaB6 scanning electron microscope at 14 kV. 


## Sample origin and sequencing of Constancias et al. 2014

### Soil DNA extraction, pyrosequencing, and data analysis 

Using the GnsGII procedure (Plassart et al. 2012), soil microbial DNA was extracted and purified from soil samples. Bacterial density was estimated by real-time quantitative PCR of 16S rRNA gene which were then amplified with 341F and 515R primers on 1 ng of template DNA and using the Veriti® Thermal Cycler. Using 454-pyrosequencing of the 16S rRNAV3–V4 gene region (Terrat et al. 2012). The first PCR used primers F479 and R888, while the second used purified PCR products from the first one. Ten base-pair multiplex identifiers (MID) were added to the primers at 5′ position. Pyrosequencing was then carried out on a GS Junior And the data were processed using the GnS-PIPE of the GenoSol platform (Terrat et al. 2012). Raw sequences were discarded if they had ambiguous base calls, length less than 350 nt, or if they contained any ambiguity (Ns). Using infernal alignments that use secondary structure of 16S rRNA gene Sequences, sequences that clustered at 95 % sequence similarity into operational taxonomic units (OTU) were identified. A filter for potential sources (chimeras, sequencing errors, OTU overestimation) of errors was applied. To compare the datasets, quality sample reads were reduced by random selection to the lowest dataset, i.e., 2,060 randomly quality-selected sequences per sample. High-quality reads were used for taxonomy-based analysis using USEARCH against reference databases from the Silva (Quast et al. 2013), with sequences longer than 500 nucleotides. Using UniFrac we compared the sampled bacterial communities (Lozupone and Knight 2005).  UniFrac community analyses were based on a phylogenetic tree computed with FastTree using the most abundant read to represent each OTU. A weighted UniFrac distance measure was then realized to compute a dendrogram by using the PycoGent package (Knight et al. 2007). I found the raw data sets On NCBI with EBI database system tags using project accession number PRJEB4553.Differences between physicochemical characteristics, density, and diversity in bacterial communities were tested by ANOVA, multiple paired comparisons with P <0.05. Using the heatmap.2 function the heatmap figure was built and the gplots R package was used for the relative abundance values of the most dominant bacterial genus across the samples. The mean of each taxonomic group across sample types was calculated and colored by a corresponding Z score.

## Computational

These are the methods you used to do your bioinformatic analyses analyses. Should probably be between 0.5 and 1 pages. At a very minimum should include citations for DADA2 and phyloseq if you are doing an amplicon study, or other citations as appropriate.

# Results

## Subsections are ok in the results section too

```{r your-code-here}
# Add code chunks as needed for your analyses
# For most analyses, I would recommend splitting the
# intensive computational part into a seperate R script file
# and then just load your libraries and the data object here
# using the `load()` function. Ask me for clarification if this is
# unclear.
```

```{r load-packages-and-data, message = FALSE}
# load general-use and special packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("citr")
library("dada2")
library("seqinr")
library("mctoolsr")
library("phyloseq")
load("output/phyloseq_obj.Rdata")
# load data produced from analysis scripts using
# something like load("output/processed_data.Rdata")
```

```{r abundancies-across-the-three-types-of-land-use}
phyloseq_obj_abundancies <- subset_samples(phyloseq_obj, # a new phyloseq object
                                  current_land_use == "grassland" |
                                    current_land_use == "till cropland" |
                                    current_land_use == "no till cropland")
plot_bar(phyloseq_obj_abundancies, x = "current_land_use",
          title = "Sequence abundance of current land use") +
  xlab("Current land use")

```

```{r abundance-genus-labeled}
plot_bar(phyloseq_obj, x = "current_land_use", fill = "Genus") +
  theme(legend.position = "none")
```

```{r sequence-abundance-filtered}
phyloseq_abundances_f <- subset_samples(phyloseq_obj, # new object
                                    current_land_use == "grassland" |
                                    current_land_use == "till cropland" |
                                    current_land_use == "no till cropland")

phyloseq_table <- psmelt(phyloseq_abundances_f) %>%
filter(Abundance < 600)
ggplot(phyloseq_table, aes(x = current_land_use, y = Abundance)) +
  geom_col() + ggtitle("Sequence abundance of current land use") +
xlab("Current land use")
```



```{r shannon-bw}
plot_richness(phyloseq_obj,
               x = "current_land_use",
               measures = c("Shannon", "Simpson"),
              title = "Shannon and Simpson index for current land use") +
   geom_jitter(width = 0.2) +
   theme_bw()
```

```{r shannon-simpson-color}
plot_richness(phyloseq_obj,
               x = "current_land_use",
               measures = c("Shannon", "Simpson"),
              color = ("Library_Name"),
title = ("Shannon and Simpson index for current land use")) +
   geom_jitter(width = 0.2)
```
```{r Shannon-color}
plot_richness(phyloseq_obj,
               x = "current_land_use",
               measures = c("Shannon"),
              color = ("Library_Name"),
title = ("Shannon and Simpson index for current land use")) +
   geom_jitter(width = 0.2)
```

```{r abundance-per-sample-by-kingdom}
plot_bar(phyloseq_obj, fill = "Kingdom") +
  ggtitle("Abundance of sequences per sample by Kingdom") +
  theme(legend.position = "bottom")
```

```{r abundance-per-sample-by-phylum}
plot_bar(phyloseq_obj, fill = "Phylum") +
  ggtitle("Abundance of sequences per sample by Phylum") +
  theme(legend.position = "bottom")
```

```{r abundance-per-sample-by-class}
plot_bar(phyloseq_obj, fill = "Class") +
  ggtitle("Abundance of sequences per sample by Class") +
  theme(legend.position = "bottom")
```

```{r abundance-per-sample-by-order}
plot_bar(phyloseq_obj, fill = "Order") +
  ggtitle("Abundance of sequences per sample by Order")
```

```{r genus-frequency-table}
table(tax_table(taxa)[, "Genus", exclude = NA]) %>%
  kable(col.names = c("Genus", "Frequency"))
```

```{r order-frequency-table}
table(tax_table(taxa)[, "Order"], exclude = NULL) %>%
kable(col.names = c("Order",
                    "Frequency"))
```

```{r class-frequency-table}
table(tax_table(taxa)[, "Class", exclude = NA]) %>%
kable(col.names = c("Class",
                    "Frequency"))
```

```{r phylum-frequency-table}
table(tax_table(taxa)[, "Phylum", exclude = NA]) %>%
kable(col.names = c("Phylum",
                    "Frequency"))
```


```{r kingdom-frequency-table}
table(tax_table(taxa)[, "Kingdom", exclude = NA]) %>%
kable(col.names = c("Kingdom",
                    "Frequency"))
```

```{r testing}
plot_bar(phyloseq_obj,
         x = "current_land_use",
         fill = "Genus") +
  geom_bar(aes(color = Genus,
               fill = Genus),
           stat = "identity",
           position = "stack") +
  theme(legend.position = "none")
```

```{r ph}
by_ph_current_land_use <- melted_phyloseq %>%
  group_by(current_land_use, pH) %>%
summarize(meanpH = mean(sumpH))
ggplot(by_pH_current_land_use, aes(x = current_land_use,
         y = meanSumpH)) +
  geom_col() +
 ggtitle("Comparing mean pH of samples by current land use")
```
In addition to a minimum of 5-8 figures/tables (and associated captions), you should include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found, but you don't interpret it - that happens in the discussion. 2-3 pages.

# Discussion

Add around 3-4 pages interpreting your results and considering future directions one might take in analyzing these data.

# Sources Cited
