---
title: "Final Report: Your Title Here"
author: "Derek Newberger"
date: "November, 26 2018"
output:
    github_document: default
bibliography: references.bib
csl: bioinformatics.csl
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Since their initial discovery, microbes have been captivating audiences. The individual most known who bring microbes to focus was Anton van Leeuwenhoek, whose infectious enthusiasm of what lived between one’s teeth or under one’s finger nails captured the interest of those in even highest social ranks [@karamanou2010anton]. What a thrill it was to see this “invisible” world in constant motion, a world that lived right under and inside our noses. But once the curtain was lifted and the initial thrill died down, it became apparent that although these microbes had been living out of site, they had always been drastically affecting our lives. Diseased and infectious, contagious and vivacious, cultivated fear. Our world waged war upon theirs. We continue this practice, but we are continually gaining a better understanding of these microorganisms. We now consume probiotics [@goldin1992probiotics], and the consequences of antibiotics is becoming more apparent as we see that it is more important to maintain a healthy microbiome and how bacterial resistance can create even larger issues [@acar1997consequences]. There are numerous estimates of how many microorganisms reside in the human body. Estimates of how many microorganisms live in our body (in a 70 kg individual) range from 38 to 100 trillion, which is something spectacular since it is estimated that there is approximately 30 trillion human cells [@sender2016revised]. It is important to keep in mind the typical size difference between eukaryotic and prokaryotic cells (though not all microorganisms are prokaryotic). However, estimates of the abundance of microorganisms in soil are even greater.  It is estimated that in one gram of soil can host 10 billion microorganisms and thousands of different species [@constancias2014microscale]. Using the previous estimate in humans, per 70 kg of soil there is 700 trillion microorganisms, a much larger number no matter the given estimate. Some individuals dedicate more time than others to interact with their local soil, but we all eat. I would like for you to consider how our current agricultural practices influence the bacterial diversity in our crop land. 
The study of interest is on the effects of tilling on the bacterial density and diversity in cropland.  Florentin Constancias and collaborators were interested in how soil microbes, which play a major role in forming organic matter in the soil that is utilized by crops as fertilizers, are affected by the current agricultural practice of tilling. Their goal was to describe how this high soil disturbance practice impacts bacterial abundance and diversity, but one a microenvironment scale of micrometers to millimeters, in a time when previous studies had focused on the macroscale of meters to kilometers. They looked at density and diversity in crop soil under grassland, tillage plots, and no tillage (stubble plowed) plots [@constancias2014microscale]. I wish to review the results of this article by attempting to replicate their results and thus support their conclusions. I will focus on their larger take home messages as receive a better understanding in what they mean by stating that the difficulties in characterizing indigenous microbial populations have reduced. I have no doubt that with next generation sequencing (NGS) that these issues have been reduced but am curious to see how many sequences that the Ribosomal database fails to identify. The authors used a curated database called Silva, and it will be interesting to see how the results from dada using the ribosomal database project (RDP) compares in regard to taxa profiling. Additionally, I wish to utilize their data on pH level. In their introduction they mentioned the importance that pH has on microorganisms three different times in that soil characteristics like pH drive microbial abundance and diversity (unlike climate and geomorphologies [@dequiedt2011biogeographical; @fierer2006diversity], heterogeneous and nonrandom distribution of the abundance and diversity of soil microbial communities influence soil characteristics like pH [@constancias2014microscale; @zinger2011contrasting], and soils provide a mosaic of microenvironments differing in pH [@mummey2006spatial; @ranjard2001quantitative]). I also want to dive a little bit deeper in the question of nitrogen fixation, by looking if and where these bacteria are located, if different crop intensities reveal a pattern, and if pH has a clear influence on these bacteria from the data of this study. We will also try to follow nitrogen. For this article, terms and concepts that are relevant to understanding the article itself and this analysis will be defined and organized by abiotic and biotic factors. 
Relevant abiotic factors in this study include characteristics of soil and chemistry of the organic compound within it. The study placed sample weights into three main categories, macroaggregates (250-2,000 μm and 63-250 μm), microaggregates (20-63μm and 2-20 μm), and clay particles (2 μm) [@constancias2014microscale]. In its broadest sense, a macroaggregates are simply relatively large particles. Similar studies have described macroaggregates as 250-2,000 μm clumps which form from fresh residues since it is a carbon source for microbes and encourages the production of micro-derived binding agents [@six2000soil]. Microaggregates are formed within the much larger macroaggregates and typically maintains stability even if the macroaggregate it is inside, breaks apart. Both of these structures influence the C-N availability The study defined clay particles as particles, lower than 2 μm, which is in line with most estimates and although not explicitly defined, the silty-clay particles can be defined as soil particles that are intermixed with particles between 2-50 μm [@six2000soil:1]. Bulk is soil outside the rhizosphere and is not penetrated by plant roots. The rhizosphere contains more natural organic compounds and micro populations than bulk soil [@blake1986bulk]. Calcisol, s a soil with a substantial secondary accumulation of lime. Calcisols are widespread in arid and semi-arid environments with a substantial secondary accumulation of lime (calcium oxide). Though not the most suitable for agriculture, calcisol environments can be irrigated, drained (prevents salinization) and fertilized, to become a highly productive [@ibanez2015there; @meeus1995pan]. Not being defined in the article, it is interesting to note that this probably means that a significant amount of fertilizer is used for these plots, bringing in a different batch of microbes. Additionally, an important micronutrient if calcium, which may encourage the growth of some bacteria over others [@thys2000micronutrients]. 

The biotic factors in the study and my analysis consist which bacteria we observed to live in the soil and agricultural techniques applied by humans to instigate biological and chemical change of the soil. Proteobacteria is a phylum of Gram-negative bacteria that include pathogens and bacteria that induce nitrogen fixation in symbiosis with plants [@madigan1997brock].  A class with in Proteobacteria is Alphaproteobacteria, whose functions varies from phototrophy, metabolizing C1-compounds, and symbionts of plants [@garrity2005class]. Rhizobiales are an order of Gram-negative under Alphaproteobacteria and are nitrogen fixation symbiotes composed of four families, Bradyrhizobiaceae, Hyphomicrobiaceae, Rhizobiaceae, and Phyllobacteriaceae [@brenner2015classification]. Actinobacteria is a phylum of Gram-positive bacteria, they help to decompose organic matter for plant uptake and are primarily soil bacteria who can also potentially fix nitrogen [@ghai2011metagenomics]. Acidiobacteria is a phylum and Acidobacteria GP6 (a class) and subgroups’ presence correlate with low values of soil pH in the northern hemisphere [@catao2014soil]. Gaiella (occulta) has been isolated from deep mineral water in Portugal [@albuquerque2011gaiella]. Pseudomonas is a genus that can be a human pathogen, soil bacterium, or plant growth-promotor [@euzeby1997list]. Massilia is a genus in the rhizosphere that colonizes plant roots to promote growth development [@berg2014unraveling]. There are plenty more taxa that could be described, but those are the ones that will be focused on. Tillage is generally described as having the greatest impact on biological properties with a strong physical disturbance of soil, leading to a decrease of soil macroaggregate content [@oorts2007determinants]. Temperate Grasslands can characterize about 20% of the land in Europe, which include France where the study took place. They can be nutrient-rich due to decay of deep branched grass roots and about 47% of grasslands have been converted into agricultural or urban developments [@soussana2004carbon], much like the crop plots in the study. They describe it as unpastured meadow plot frequently mowed since 2002 [@constancias2014microscale]. The study used two cultivated plots under the same crop rotation of rape, wheat, and barley [@constancias2014microscale]. The plot labeled as “not tilled” was only stubble ploughed which means that the tilling was shallow and performed immediately before sowing to loosen the surface and damage insect cocoons that reside of the superficial layer of soil [@wheatley1995influence]. Lastly, plants require nitrogen because it is essential in chloroplast formation [@turner1994assessing].
In conclusion, we have discussed the study in question, the intent of the analysis, and the abiotic and biotic factors in depth to provide sufficient background for the analysis. 

# Methods

## Study design of Constancias et al. 2014

### Field soil sampling:

During springtime in Burgundy, France in the agricultural landscape of Fénay, soils were sampled from three plots of soil characterized by silty clay calcisol. The agricultural practices on the grassland plot, the tilled plot, and the stubble ploughed plot, of the past eight years were monitored. Each plot had minor differences in their parameters. For sampling, a geological reference point of one meter was used and soil cores were from the surface horizon (0-20 cm) and pooled. After being sieved (2mm), the samples were air dried, and then stored at 4 C until the fraction procedure.

### Soil fractionation procedure:

They used a gentle physical soil fraction method (Chotte et al. 2002), to isolate the soil microenvironments by size. A subsample of 50 g of 2-mm was shaken in 250 mL of Milli-Q water for 3 h. Using the two sieves (250- and 63-μm) particles and water-stable aggregates were separated by wet sieving in approximately 3 cm of water 60 times during 2 min.  The 2,000–250- and 250–63-μm fractions were macroaggregates associated to coarse and fine sands, respectively. Stoke’s law was used to separate the microaggregates with particles of silt and loam by sedimentation and siphoning. Water was removed as clay fractions of <2 μm were recovered by centrifuging at 11,000g, 4 °C for 20 min. Samples were then freeze-dried. This procedure was carried out in replicates. Replicates were lyophilized and stored at −40 °C, and the three subsamples were used for physicochemical and microbiological analysis.

### Soil physicochemical analysis

The lyophilized fractions were weighed, organic carbon and total nitrogen contents of samples were characterized according to NF ISO 10694 and 13878.  Soil texture of bulk soils was determined according to NF X 31–107. Size distribution of the particles and aggregates of each fraction was verified by laser granulometry (Mastersizer 2000, Malvern Ltd). Using adhesive graphite tabs, aluminum stubs gold-palladium (12–15 nm thick), Polaron SC 7620 Mini Sputter Coater, samples were observed with a Philips XL-30ESEM LaB6 scanning electron microscope at 14 kV. 


## Sample origin and sequencing of Constancias et al. 2014

### Soil DNA extraction, pyrosequencing, and data analysis 

Using the GnsGII procedure, soil microbial DNA was extracted and purified from soil samples. Bacterial density was estimated by real-time quantitative PCR of 16S rRNA gene which were then amplified with 341F and 515R primers on 1 ng of template DNA and using the Veriti® Thermal Cycler. Using 454-pyrosequencing of the 16S rRNAV3–V4 gene region. The first PCR used primers F479 and R888, while the second used purified PCR products from the first one. Ten base-pair multiplex identifiers (MID) were added to the primers at 5′ position. Pyrosequencing was then carried out on a GS Junior And the data were processed using the GnS-PIPE of the GenoSol platform. Raw sequences were discarded if they had ambiguous base calls, length less than 350 nt, or if they contained any ambiguity (Ns). Using infernal alignments that use secondary structure of 16S rRNA gene Sequences, sequences that clustered at 95 % sequence similarity into operational taxonomic units (OTU) were identified. A filter for potential sources (chimeras, sequencing errors, OTU overestimation) of errors was applied. To compare the datasets, quality sample reads were reduced by random selection to the lowest dataset, i.e., 2,060 randomly quality-selected sequences per sample. High-quality reads were used for taxonomy-based analysis using USEARCH against reference databases from the Silva, with sequences longer than 500 nucleotides. Using UniFrac we compared the sampled bacterial communities.  UniFrac community analyses were based on a phylogenetic tree computed with FastTree using the most abundant read to represent each OTU. A weighted UniFrac distance measure was then realized to compute a dendrogram by using the PycoGent package. I found the raw data sets On NCBI with EBI database system tags using project accession number PRJEB4553.Differences between physicochemical characteristics, density, and diversity in bacterial communities were tested by ANOVA, multiple paired comparisons with P <0.05. Using the heatmap.2 function the heatmap figure was built and the gplots R package was used for the relative abundance values of the most dominant bacterial genus across the samples. The mean of each taxonomic group across sample types was calculated and colored by a corresponding Z score (constancias 2014).

## Computational

These are the methods you used to do your bioinformatic analyses analyses. Should probably be between 0.5 and 1 pages. At a very minimum should include citations for DADA2 and phyloseq if you are doing an amplicon study, or other citations as appropriate.

# Results

## Subsections are ok in the results section too

```{r your-code-here}
# Add code chunks as needed for your analyses
# For most analyses, I would recommend splitting the
# intensive computational part into a seperate R script file
# and then just load your libraries and the data object here
# using the `load()` function. Ask me for clarification if this is
# unclear.
```

```{r load-packages-and-data, message = FALSE}
# load general-use and special packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("citr")
library("dada2")
library("seqinr")
library("mctoolsr")
library("phyloseq")
load("output/phyloseq_obj.Rdata")
# load data produced from analysis scripts using
# something like load("output/processed_data.Rdata")
```

```{r abundancies-across-the-three-types-of-land-use}
phyloseq_obj_abundancies <- subset_samples(phyloseq_obj, # a new phyloseq object
                                  current_land_use == "grassland" |
                                    current_land_use == "till cropland" |
                                    current_land_use == "no till cropland")
plot_bar(phyloseq_obj_abundancies, x = "current_land_use",
          title = "Sequence abundance of current land use") +
  xlab("Current land use")

```

```{r sequence-abundance-filtered}
phyloseq_abundances_f <- subset_samples(phyloseq_obj, # new object
                                    current_land_use == "grassland" |
                                    current_land_use == "till cropland" |
                                    current_land_use == "no till cropland")
phyloseq_table <- psmelt(phyloseq_abundances_f) %>%
filter(Abundance < 600)
ggplot(phyloseq_table, aes(x = current_land_use, y = Abundance)) +
  geom_col() + ggtitle("Sequence abundance of current land use") +
xlab("Current land use")
```

```{r Shannon-color}
plot_richness(phyloseq_obj,
               x = "current_land_use",
               measures = c("Shannon"),
              color = ("Library_Name"),
title = ("Shannon index for current land use"))
```

```{r shannon-chao1-color}
plot_richness(phyloseq_obj,
               x = "current_land_use",
               measures = c("Shannon", "Chao1"),
              color = "Library_Name",
              title = "Shannon and Simpson index for current land use")
```


```{r abundance-per-sample-by-kingdom}
plot_bar(phyloseq_obj, fill = "Kingdom") +
  geom_bar(aes(color = Kingdom,
               fill = Kingdom),
           stat = "identity",
           position = "stack") +
  ggtitle("Abundance of sequences per sample by Kingdom") +
  theme(legend.position = "bottom")
```

```{r kingdom-frequency-table}
as.data.frame(tax_table(phyloseq_obj)) %>%
group_by(Kingdom) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(3) %>%
  kable(col.names = c("Kingdom",
                    "Frequency"))
```

```{r abundance-per-sample-by-phylum}
plot_bar(phyloseq_obj, fill = "Phylum") +
    geom_bar(aes(color = Phylum,
               fill = Phylum),
           stat = "identity",
           position = "stack") +
  ggtitle("Abundance of sequences per sample by Phylum") +
  theme(legend.position = "bottom")
```

```{r phylum-frequency-table}
as.data.frame(tax_table(phyloseq_obj)) %>%
group_by(Phylum) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(10) %>%
  kable(col.names = c("Phylum",
                    "Frequency"))
```

```{r abundance-per-sample-by-class}
plot_bar(phyloseq_obj, fill = "Class") +
  geom_bar(aes(color = Class,
               fill = Class),
           stat = "identity",
           position = "stack") +
  ggtitle("Abundance of sequences per sample by Class") +
  theme(legend.position = "bottom")
```

```{r class-frequency-table}
as.data.frame(tax_table(phyloseq_obj)) %>%
group_by(Class) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(10) %>%
  kable(col.names = c("Class",
                    "Frequency"))
```

```{r family-frequency-table}
as.data.frame(tax_table(phyloseq_obj)) %>%
group_by(Family) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(24) %>% # Stop at 24 since that is when we see Rhizobiaceae
  kable(col.names = c("Family",
                    "Frequency"))
```

Figure 1 All four nitrogen bacteria are present
```{r abundance-per-sample-by-order}
plot_bar(phyloseq_obj, fill = "Order") +
  geom_bar(aes(color = Order,
               fill = Order),
           stat = "identity",
           position = "stack") +
  theme(legend.position = "none")
  ggtitle("Abundance of sequences per sample by Order")
```

```{r order-frequency-table}
as.data.frame(tax_table(phyloseq_obj)) %>%
group_by(Order) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(10) %>%
  kable(col.names = c("Order",
                    "Frequency"))
```

```{r abundance-per-sample-by-genus}
plot_bar(phyloseq_obj, fill = "Genus") +
  geom_bar(aes(color = Genus,
               fill = Genus),
           stat = "identity",
           position = "stack") +
   theme(legend.position = "none")
  ggtitle("Abundance of sequences per sample by Genus")
```

```{r genus-frequency-table}
as.data.frame(tax_table(phyloseq_obj)) %>%
  group_by(Genus) %>%
  tally() %>%
  arrange(desc(n)) %>%
  head(31) %>%
  kable(col.names = c("Genus", "Frequency"))
```

```{r abundance-land-use-genus}
plot_bar(phyloseq_obj,
         x = "current_land_use",
         fill = "Genus") +
  geom_bar(aes(color = Genus,
               fill = Genus),
           stat = "identity",
           position = "stack") +
  theme(legend.position = "bottom")
```

In addition to a minimum of 5-8 figures/tables (and associated captions), you should include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found, but you don't interpret it - that happens in the discussion. 2-3 pages.

# Discussion

Add around 3-4 pages interpreting your results and considering future directions one might take in analyzing these data.

# Sources Cited
